### ISSUE TITLE
Verify and enrich session on server-side

### ISSUE THREAD

**Issue by tobiasdiez** [atinux/nuxt-auth-utils#203]
Often one needs to check if a session is still valid on the server-side. For example, one may want to present the user with an option to logout on all devices, which then should invalidate all existing sessions.

For this, one needs to query for every request the a sever-side session storage (and perhaps update it). Currently, this is relatively hard to implement. A few suggestions to make this easier:
1. Implement a `useRawSession`  method to directly access the h3 session (in other words make https://github.com/atinux/nuxt-auth-utils/blob/e0396255b2ebc6b3ef89b9dd62f0eb0fbf345389/src/runtime/server/utils/session.ts#L96 public) - alternatively, add the `sessionId` to the public interface of `UserSession`
2. Implement a server middleware that calls a hook where devs can check the validity of the `sessionId` (and update timestamps like "last active" etc)
3. ...and perhaps return some data that one would like to associate with the current session on all server routes but don't directly store in the user cookie (e.g. permissions)

Point two and three might look like:
```ts
export default defineEventHandler(async (event) => {
  const session = await _useSession(event)
  // session.id is not a good way to check if there is a session as it will always be set by h3 (except if one calls clearSession)
  // use session.data instead, with the convention that if it is an empty object, there is no session ?
  if (session.id && Object.keys(session.data).length !== 0) {
    try {
      // Check if session is valid
      const info = await hooks.call('validateSession', {id: session.id, data: session.data})
      // Is there a better to store data in the session object?
      session.server = info
    } catch {
      // Clear session
      await session.clear()
    }
  }
})
```

**Comment by atinux** [comment]
Did you take a look at the `fetch` hook (https://github.com/atinux/nuxt-auth-utils?tab=readme-ov-file#extend-session) to already implement this logic?

**Comment by timyourivh** [comment]
What I did with @sidebase/nuxt-auth was reverse engineer the way they encrypt the cookie.
As I'm currently migrating the authentication to this package meant that I needed to do the same.
They make use of h3's sessions. "h3" is the server Nitro and thus Nuxt 3 is built upon. h3 encrypts their session data using using [this "seal" method](https://github.com/unjs/h3/blob/7324eeec854eecc37422074ef9f2aec8a5e4a816/src/utils/internal/iron-crypto.ts#L64-L90) which makes clear they use the [iron method popularized by Hapi](https://hapi.dev/module/iron/).

So there you have it. If you want to encrypt/decrypt, seal/unseal the data on a server you'd have to use iron to do so. 

You could write all the logic yourself, but here are some packages that can do it for you: 

#### JS:
- Iron webcrypto: https://www.npmjs.com/package/iron-webcrypto
- Hapi iron: https://www.npmjs.com/package/@hapi/iron

#### PHP:
- iron-crypto-php: https://github.com/shawm11/iron-crypto-php

**Comment by tobiasdiez** [comment]
Yes, I did. But this is not run on server-routes (right?) and doesn't provide the sessionId.

**Comment by atinux** [comment]
What about creating a server util to wrap `requireUserSession(event)`?

```ts
// server/utils/session.ts
export async function requireValidUserSession(event) {
  const session = await requireUserSession(event)

  // Do your logic here or throw createError(...)
  // return the extended session
  return session
})
```

What we can do is to expose the `session.id`, or you can do you own with:

```ts
await setUserSession({ id: randomUUID(), user: { ... } })
```

**Comment by tobiasdiez** [comment]
Yes, the approach via `requireValidUserSession` would work as well and I'm using something like this in https://github.com/JabRef/JabRefOnline/blob/main/server/middleware/validateSession.ts. I just thought that it is a very common pattern and thus would have liked the nuxt-auth module to expose such a hook.

**Comment by ineshbose** [comment]
> What we can do is to expose the `session.id`, or you can do you own with:
> 
> ```ts
> await setUserSession({ id: randomUUID(), user: { ... } })
> ```

I was looking to access the `session.id` -- while also putting our own is also one solution, could it be that it may create some confusion between the userSession ID vs session ID? While exposing `session.id` could make the DX a bit easier

**Comment by atinux** [comment]
Nice suggestion @ineshbose 

I now forward the session id from h3 with #338, that you can overwrite also using `setUserSession({ id: 'my-id' })`
